programmers/신규-아이디-추천.js

```diff
@@ -0,0 +1,21 @@
+function solution(new_id) {
+    new_id = new_id
+        .toLowerCase()
+        .replace(/[^\w\.\-]/g, '')
+        .replace(/[\.]{2,}/g, '.')
+        .replace(/^\./, '')
+        .replace(/\.$/, '');
+    
+    if (!new_id)  new_id = "a";
+    if (new_id.length >= 16) {
+        new_id = new_id
+            .slice(0, 15)
+            .replace(/\.$/, '');
+    }
```

Comment:
굳이 !new_id 알아보고 new_id = 'a'할 필요없이 바로 `.replace(/^$/, 'a')` 를 추가하면 됨
또, new_id.length >= 16 조건 굳이 확인할 필요 없이 처음부터 걍 slice(0, 15)로 짜르면 됨
```suggestion
function solution(new_id) {
    new_id = new_id
        .toLowerCase() //1단계
        .replace(/[^\w\.\-]/g, '') //2단계
        .replace(/[\.]{2,}/g, '.') //3단계
        .replace(/^\./, '') //4단계 맨 앞 '.' 제거
        .replace(/\.$/, '') //4단계 맨 뒤 '.' 제거
        .replace(/^$/, 'a') //5단계
        .slice(0, 15).replace(/\.$/, ''); //6단계
```